python/ray/air/checkpoint.py

```diff
@@ -246,6 +246,37 @@ def _metadata(self, metadata: _CheckpointMetadata):
         for attr, value in metadata.checkpoint_state.items():
             setattr(self, attr, value)
 
+    @property
+    def path(self) -> Optional[str]:
+        """Return path to checkpoint, if available.
+
+        This will return a URI to cloud storage if this checkpoint is
+        persisted on cloud, or a local path if this checkpoint
+        is persisted on local disk and available on the current node.
+
+        In all other cases, this will return None.
+
+        Example:
+
+            >>> from ray.air import Checkpoint
+            >>> checkpoint = Checkpoint.from_uri("s3://some-bucket/some-location")
+            >>> assert checkpoint.path == "s3://some-bucket/some-location"
+            >>> checkpoint = Checkpoint.from_dict({"data": 1})
+            >>> assert checkpoint.path == None
+
+        Returns:
+            Checkpoint path if this checkpoint is reachable from the current node (e.g.
+            cloud storage or locally available directory).
+
+        """
+        if self._uri:
+            return self._uri
+
+        if self._local_path:
+            return self._local_path
+
+        return None
```

Comment:
Just to understand, a Checkpoint that's *both* uri backed and local path backed happens only in the following case?

1. `ckpt = Checkpoint.from_uri(...)` --> sets uri
2. `ckpt.to_directory(...)` --> sets local path


`ckpt = Checkpoint.from_directory() -> ckpt.to_uri()` does not seem to do the same. (only directory backed in this case.)

Reply:
```suggestion
        return self._uri or self._local_path
```

Reply:
There should be no way that both `self._uri` and `self._local_path` are set at the same time. If a `file://` path is passed to `Checkpoint.from_uri`, the `local_path` property is set instead

Reply:
I see, got it. uri backed checkpoint is always only uri backed even if it writes to a local dir later on.